### PR TITLE
2023-05-20-sbiobertresolve_icd10cm_augmented_en

### PR DESCRIPTION
**Pull request by jsl-models**

### CODE REVIEW

File: docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+supported: true
+annotator: SentenceEntityResolverModel
+article_header:
+  type: cover
+use_language_switcher: "Python-Scala-Java"
+---
+
+## Description
+
+This model maps extracted medical entities to ICD10-CM codes using `sbiobert_base_cased_mli` Sentence Bert Embeddings. Also, it has been augmented with synonyms for making it more accurate.
```
Comment (muhammetsnts) [inline docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md:21]: Add "and returns the official resolution text within the brackets."

File: docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+[Download](https://s3.amazonaws.com/auxdata.johnsnowlabs.com/clinical/models/sbiobertresolve_icd10cm_augmented_en_4.4.2_3.0_1684591054927.zip){:.button.button-orange.button-orange-trans.arr.button-icon.hidden}
+[Copy S3 URI](s3://auxdata.johnsnowlabs.com/clinical/models/sbiobertresolve_icd10cm_augmented_en_4.4.2_3.0_1684591054927.zip){:.button.button-orange.button-orange-trans.button-icon.button-copy-s3}
+
+## How to use
+
+
+
+<div class="tabs-box" markdown="1">
+{% include programmingLanguageSelectScalaPythonNLU.html %}
+```python
```
Comment (muhammetsnts) [inline docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md:39]: Add indentations

File: docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+.setOutputCol("resolution")\
+.setDistanceFunction("EUCLIDEAN")
+
+nlpPipeline = Pipeline(stages=[document_assembler, sentence_detector, tokenizer, word_embeddings, clinical_ner, ner_converter, chunk2doc, sbert_embedder, icd10_resolver])
+
+data_ner = spark.createDataFrame([["A 28-year-old female with a history of gestational diabetes mellitus diagnosed eight years prior to presentation and subsequent type two diabetes mellitus (T2DM), one prior episode of HTG-induced pancreatitis three years prior to presentation, associated with acute hepatitis, and obesity with a body mass index (BMI) of 33.5 kg/m2, presented with a one-week history of polyuria, polydipsia, poor appetite, and vomiting. Two weeks prior to presentation, she was treated with a five-day course of amoxicillin for a respiratory tract infection."]]).toDF("text")
+
+results = nlpPipeline.fit(data_ner).transform(data_ner)
+```
+```scala
```
Comment (muhammetsnts) [inline docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md:87]: Add indentations

File: docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+
+val clinical_ner = MedicalNerModel.pretrained("ner_clinical", "en", "clinical/models")
+.setInputCols(Array("sentence","token","embeddings"))
+.setOutputCol("ner")
+
+val ner_converter = NerConverter()
+.setInputCols(Array("sentence","token","ner"))
+.setOutputCol("ner_chunk")
+.setWhiteList(Array('PROBLEM'))
```
Comment (muhammetsnts) [inline docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md:114]: Use " here

File: docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
++-------------------------------------+-------+------------+-------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------+
+|        gestational diabetes mellitus|PROBLEM|       O24.4|gestational diabetes mellitus [gestational diabetes mellitus], gestational diabetes mellitus [gestational... |O24.4, O24.41, O24.43, Z86.32, Z87.5, O24.31, O24.11, O24.1, O24.81...|
+|subsequent type two diabetes mellitus|PROBLEM|      O24.11|pre-existing type 2 diabetes mellitus [pre-existing type 2 diabetes mellitus], disorder associated with t... |O24.11, E11.8, E11, E13.9, E11.9, E11.3, E11.44, Z86.3, Z86.39, E11...|
+|                   an acute hepatitis|PROBLEM|         B15|acute hepatitis a [acute hepatitis a], acute hepatitis [acute hepatitis], acute infectious hepatitis[acu...  |B15, K72.0, B17.9, B17.2, Z03.89, B15.9, K75.2, K71.2, B19.9, B16, ...|
```
Comment (muhammetsnts) [inline docs/_posts/Ahmetemintek/2023-05-20-sbiobertresolve_icd10cm_augmented_en.md:152]: result is not good for this chunk, update text.

**Review by muhammetsnts — CHANGES_REQUESTED** [review]

**Review by muhammetsnts — APPROVED** [review]
💯👏

**Comment by vkocaman** [comment]
Good job guys !

**Pull request merged**